tests/src/Traits/MockSchemaTrait.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\Tests\graphql\Traits;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\graphql\Plugin\GraphQL\SchemaPluginManager;
+use Drupal\graphql\Plugin\GraphQL\Schemas\SchemaPluginBase;
+use Drupal\KernelTests\KernelTestBase;
+use Prophecy\Argument;
+
+trait MockSchemaTrait {
+
+  abstract function getSchemaDefinitions();
+
+  /**
+   * @var \Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $schemaManagerProphecy;
+
+  public function schemaManagerFactory() {
+    return $this->schemaManagerProphecy->reveal();
+  }
+
+  protected function registerSchemaPluginManager(ContainerBuilder $container) {
+    if ($this instanceof KernelTestBase) {
+      $this->schemaManagerProphecy = $this->prophesize(SchemaPluginManager::class);
+
+      $that = $this;
```

Comment:
Why is this required?

Reply:
The prophecy callback is bound to the prophecy object, so `$this` won't work. But perhaps there's a more elegant way around this.

Reply:
Interesting. Do they somehow overwrite the normal context (which is "$this from the place you create the closure in") through reflection?

Reply:
It uses `Closure::bind`. We could just introduce our own callback promise to avoid that.